cli/types/index.d.ts

```diff
@@ -174,757 +182,785 @@ declare namespace Cypress {
 
     /**
+     * Scroll an element into view.
+     *
+     * @see https://on.cypress.io/scrollintoview
+     */
+    scrollIntoView(options?: Partial<ScrollIntoViewOptions>): Chainable<Subject>
+
+    /**
+     * Scroll to a specific position.
+     *
+     * @see https://on.cypress.io/scrollto
+     */
+    scrollTo(position: PositionType, options?: Partial<ScrollToOptions>): Chainable<Subject>
+    scrollTo(x: number | string, y: number | string, options?: Partial<ScrollToOptions>): Chainable<Subject>
+
+    /**
+     * Select an `<option>` within a `<select>`.
+     *
      * @see https://on.cypress.io/select
      */
-    select(text: string | string[], options?: SelectOptions): Chainable
-    select(value: string | string[], options?: SelectOptions): Chainable
+    select(text: string | string[], options?: Partial<SelectOptions>): Chainable<Subject>
+    select(value: string | string[], options?: Partial<SelectOptions>): Chainable<Subject>
 
     /**
+     * Start a server to begin routing responses to `cy.route()` and `cy.request()`.
+     *
      * @see https://on.cypress.io/server
      */
-    server(options?: ServerOptions): Chainable
+    server(options?: Partial<ServerOptions>): Chainable<null>
 
     /**
+     * Set a browser cookie.
+     *
      * @see https://on.cypress.io/setcookie
      */
-    setCookie(name: string, value: string, options?: SetCookieOptions): Chainable
+    setCookie(name: string, value: string, options?: Partial<SetCookieOptions>): Chainable<Cookie>
 
     /**
+     * Create an assertion. Assertions are automatically retried until they pass or time out.
+     *
      * @see https://on.cypress.io/should
      */
-    should(chainers: string, value?: any): Chainable
-    should(chainers: string, method: string, value: any): Chainable
-    should(fn: (currentSubject?: any) => void): Chainable
+    should: Chainer<Subject>
 
     /**
+     * Get sibling DOM elements.
+     *
      * @see https://on.cypress.io/siblings
      */
-    siblings(options?: LoggableTimeoutable): Chainable
-    siblings(selector: string, options?: LoggableTimeoutable): Chainable
+    siblings<K extends keyof HTMLElementTagNameMap>(selector: K, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<HTMLElementTagNameMap[K]>>
+    siblings<E extends Node = HTMLElement>(options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+    siblings<E extends Node = HTMLElement>(selector: string, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
 
     /**
-     * @see https://on.cypress.io/spread
+     * Wrap a method in a spy in order to record calls to and arguments of the function.
+     * > Note: `.spy()` assumes you are already familiar with our guide: [Stubs, Spies, and Clocks](https://docs.cypress.io/guides/guides/stubs-spies-and-clocks.html)
+     *
+     * @see https://on.cypress.io/spy
      */
-    spread(fn: (...args: any[]) => any): Chainable
+    spy(): sinon.SinonSpy & Chainable<sinon.SinonSpy>

```

Comment:
@brian-mann The docs say `cy.spy` returns `sinon.spy`, but I also see an alias. Here I'm adding all of Chainable to the return, but I'm guessing that isn't correct. I could find the implementation to confirm.

Does `cy.spy()` and `cy.stub` have only a special non-chainable `as` method added to them and nothing else? If so, I should update this type

Reply:
Correct. They only have a single `.as` method added to them. This brings them into parity with other Cypress idioms. They show up in the Command Log highlighted with the alias, and it also enables you to use them with `cy.get('@alias')`

Here's the code.

https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/agents.coffee#L189

Reply:
FWIW there is also a `log` method which enables you to turn off Command Log logging. This is necessary in situations where the spied / stubbed function is called hundreds of times and would lock up the UI.

Reply:
Thanks. Not sure what to call the interface. Perhaps `Agent`? It looks like it returns a Spy/Stub with a chainable interface with just `.log` and `.as`.

Reply:
Fixed. I see that `withArgs` was overridden, so the `Agent` interface does the same thing (to allow the continuation of the `Agent` chainable